Failed to install clarity-cli native binary during npm install

**OS**: Kubuntu 18.04 x64
**Node version**: 10.16.0
**Npm version**: 6.9.0

**logs below:**

```
Fetching https://github.com/blockstack/clarity-js-sdk/releases/download/clarity-sdk-v0.0.4/clarity-cli-linux-x64.tar.gz
Extracting to temp dir /tmp/PWObdR
Moving /tmp/PWObdR/clarity-cli to /home/nervehammer/clarity-js-sdk/packages/clarity-native-bin/.native-bin/clarity-sdk-v0.0.4/clarity-cli
Failed to install clarity-cli native binary: Error: EXDEV: cross-device link not permitted, rename '/tmp/PWObdR/clarity-cli' -> '/home/nervehammer/clarity-js-sdk/packages/clarity-native-bin/.native-bin/clarity-sdk-v0.0.4/clarity-cli'
child_process.js:650
    throw err;
    ^

Error: Command failed: npx ts-node@^8.2.0 --project /home/nervehammer/clarity-js-sdk/packages/clarity-native-bin/tsconfig.build.json /home/nervehammer/clarity-js-sdk/packages/clarity-native-bin/src/directInstall.ts
    at checkExecSyncError (child_process.js:629:11)
    at Object.execFileSync (child_process.js:647:13)
    at tsNodeInstall (/home/nervehammer/clarity-js-sdk/packages/clarity-native-bin/postInstallScript.js:57:16)
    at Object.<anonymous> (/home/nervehammer/clarity-js-sdk/packages/clarity-native-bin/postInstallScript.js:19:3)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:829:12)
lerna info lifecycle @blockstack/clarity-native-bin@0.1.9-alpha.0~postinstall: Failed to exec postinstall script
lerna ERR! lifecycle "postinstall" errored in "@blockstack/clarity-native-bin", exiting 1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! clarity-monorepo-wrapper@0.0.0 prepare: `lerna bootstrap`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the clarity-monorepo-wrapper@0.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/nervehammer/.npm/_logs/2019-07-01T10_49_22_531Z-debug.log
```

--- a/package.json
+++ b/package.json
@@ -24,9 +24,11 @@
     "codecov-upload": "codecov"
   },
   "devDependencies": {
+    "@types/fs-extra": "^8.0.0",
     "@types/node": "^10.14.8",
     "codecov": "^3.5.0",
     "cross-env": "^5.2.0",
+    "fs-extra": "^8.1.0",
     "lerna": "^3.14.1",
     "typescript": "^3.5.1"
   }

--- a/packages/clarity-native-bin/postInstallScript.js
+++ b/packages/clarity-native-bin/postInstallScript.js
@@ -54,5 +54,5 @@ function tsNodeInstall() {
   const directInstallTsFile = path.join(__dirname, "src", "directInstall.ts");
   const tsNodeExecArgs = [tsNodePkg, "--project", tsConfigBuildFile, directInstallTsFile];
   console.log(`Running: npx ${tsNodeExecArgs.join(" ")}`);
-  childProcess.execFileSync("npx", tsNodeExecArgs, { stdio: ['pipe', process.stdout, process.stderr] });
+  childProcess.execFileSync("npx", tsNodeExecArgs, { stdio: ['pipe', process.stdout, process.stderr], shell: process.platform=="win32" });
 }

--- a/packages/clarity-native-bin/src/cargoBuild.ts
+++ b/packages/clarity-native-bin/src/cargoBuild.ts
@@ -1,4 +1,4 @@
-import * as fs from "fs";
+import * as fs from "fs-extra";
 import * as path from "path";
 import { executeCommand } from "./execUtil";
 import { getExecutableFileName, makeUniqueTempDir } from "./fsUtil";
@@ -88,7 +88,7 @@ export async function cargoInstall(opts: {
   const tempBinFilePath = path.join(tempCompileBinDir, binFileName);
 
   opts.logger.log(`Moving ${tempBinFilePath} to ${opts.outputFilePath}`);
-  fs.renameSync(tempBinFilePath, opts.outputFilePath);
+  fs.moveSync(tempBinFilePath, opts.outputFilePath);
 
   return true;
 }

--- a/packages/clarity-native-bin/src/fetchDist.ts
+++ b/packages/clarity-native-bin/src/fetchDist.ts
@@ -1,4 +1,4 @@
-import * as fs from "fs";
+import * as fs from "fs-extra";
 import fetch from "node-fetch";
 import * as os from "os";
 import * as path from "path";
@@ -119,8 +119,8 @@ export async function fetchDistributable(opts: {
   const tempBinFilePath = path.join(tempExtractDir, binFileName);
 
   opts.logger.log(`Moving ${tempBinFilePath} to ${opts.outputFilePath}`);
-  fs.renameSync(tempBinFilePath, opts.outputFilePath);
-  fs.rmdirSync(tempExtractDir);
+  fs.moveSync(tempBinFilePath, opts.outputFilePath);
+  fs.removeSync(tempExtractDir);
 
   return true;
 }

--- a/packages/clarity-native-bin/src/fsUtil.ts
+++ b/packages/clarity-native-bin/src/fsUtil.ts
@@ -1,4 +1,4 @@
-import * as fs from "fs";
+import * as fs from "fs-extra";
 import * as os from "os";
 import * as path from "path";
 import { ILogger } from "./logger";

--- a/packages/clarity-native-bin/src/index.ts
+++ b/packages/clarity-native-bin/src/index.ts
@@ -1,4 +1,4 @@
-import * as fs from "fs";
+import * as fs from "fs-extra";
 import * as path from "path";
 import { cargoInstall } from "./cargoBuild";
 import { fetchDistributable, getDownloadUrl } from "./fetchDist";

--- a/packages/clarity-native-bin/test/installFromDist.ts
+++ b/packages/clarity-native-bin/test/installFromDist.ts
@@ -1,5 +1,5 @@
 import { assert } from "chai";
-import fs from "fs";
+import fs from "fs-extra";
 import path from "path";
 import { CORE_SDK_TAG, install } from "../src";
 import { makeUniqueTempDir } from "../src/fsUtil";
@@ -35,7 +35,7 @@ describe("install via dist", () => {
       fs.unlinkSync(filePath);
     }
     if (fs.existsSync(tempDir)) {
-      fs.rmdirSync(tempDir);
+      fs.removeSync(tempDir);
     }
   });
 });

--- a/packages/clarity-native-bin/test/installFromSource.ts
+++ b/packages/clarity-native-bin/test/installFromSource.ts
@@ -1,5 +1,5 @@
 import { assert } from "chai";
-import fs from "fs";
+import fs from "fs-extra";
 import path from "path";
 import { CORE_SDK_TAG, install } from "../src";
 import { makeUniqueTempDir } from "../src/fsUtil";
@@ -9,7 +9,7 @@ describe("install via build from source", () => {
   let tempDir: string;
   let filePath: string;
 
-  before(function() {
+  before(function () {
     if (process.env.SKIP_SLOW_TESTS) {
       this.skip();
       return;
@@ -39,7 +39,7 @@ describe("install via build from source", () => {
       fs.unlinkSync(filePath);
     }
     if (fs.existsSync(tempDir)) {
-      fs.rmdirSync(tempDir);
+      fs.removeSync(tempDir);
     }
   });
 });

--- a/packages/clarity/src/providers/clarityBin/index.ts
+++ b/packages/clarity/src/providers/clarityBin/index.ts
@@ -1,4 +1,4 @@
-import fs from "fs";
+import fs from "fs-extra";
 import { CheckResult, Receipt } from "../../core";
 import { Provider } from "../../core/provider";
 import { getContractFilePath } from "../../utils/contractSourceDir";
@@ -155,7 +155,7 @@ export class NativeClarityBinProvider implements Provider {
     if (result.exitCode !== 0) {
       throw new ExecutionError(
         `Execute expression on contract failed with bad exit code ${result.exitCode}: ${
-          result.stderr
+        result.stderr
         }`,
         result.exitCode,
         result.stdout,
@@ -221,7 +221,7 @@ export class NativeClarityBinProvider implements Provider {
     if (result.exitCode !== 0) {
       throw new ExecutionError(
         `Eval expression on contract failed with bad exit code ${result.exitCode}: ${
-          result.stderr
+        result.stderr
         }`,
         result.exitCode,
         result.stdout,

--- a/packages/clarity/src/utils/fsUtil.ts
+++ b/packages/clarity/src/utils/fsUtil.ts
@@ -1,4 +1,4 @@
-import * as fs from "fs";
+import * as fs from "fs-extra";
 import * as os from "os";
 import * as path from "path";
 

--- a/packages/clarity/test/contractInterfaceTests.ts
+++ b/packages/clarity/test/contractInterfaceTests.ts
@@ -1,5 +1,5 @@
 import { assert } from "chai";
-import * as fs from "fs";
+import * as fs from "fs-extra";
 import * as path from "path";
 import {
   ContractInterface,

